### PR TITLE
Drop py2 support from atomic-reactor

### PR DESCRIPTION
**Pull request by SalatskySal**
# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- [x] JSON/YAML configuration changes are updated in the relevant schema
- [x] Changes to metadata also update the documentation for the metadata
- [x] Pull request has a link to an osbs-docs PR for user documentation updates
- [x] New feature can be disabled from a configuration file

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -55,7 +46,7 @@ def _install_requirements():
         'console_scripts': ['atomic-reactor=atomic_reactor.cli.main:run'],
     },
     packages=find_packages(exclude=["*.tests", "*.tests.*", "tests.*", "tests"]),
-    install_requires=_install_requirements(),
+    install_requires=_get_requirements('requirements.txt'),
     package_data={'atomic_reactor': ['schemas/*.json']},
     data_files=data_files.items(),
```
Comment (MartinBasti) [inline setup.py:51]: `python_requires='>=3.6, <4',` I'd also add this into setup.py

File: test.sh [hunk 2]
```diff
@@ -66,44 +60,23 @@ function setup_osbs() {
...
-  if [[ $OS == "centos" ]]; then
-    # RPM install dependencies for unit tests, as check is disabled for rhel
-    $RUN yum install -y python-flexmock python-six \
-                        python-backports-lzma \
-                        python-backports-ssl_match_hostname \
-                        python-responses \
-                        PyYAML \
-                        python-requests python-requests-kerberos # OSBS dependencies
-  else
+  $RUN "${BUILDDEP[@]}" --define "with_python3 1" -y atomic-reactor.spec
```
Comment (MartinBasti) [inline test.sh:64]: `with_python3`  IMO this should be removed in commit where py3 support is removed from scpecfile

**Comment by MartinBasti** [comment]
Can we address skipped tests in centos8 job?
```
=========================== short test summary info ============================
7881
SKIPPED [5] tests/plugins/test_flatpak_create_dockerfile.py:91: libmodulemd not available
7882
SKIPPED [7] tests/plugins/test_flatpak_create_oci.py:467: libmodulemd not available
7883
SKIPPED [1] tests/plugins/test_flatpak_create_oci.py:700: libmodulemd not available
7884
SKIPPED [5] tests/plugins/test_flatpak_update_dockerfile.py:172: libmodulemd not available
7885
SKIPPED [1] tests/plugins/test_flatpak_update_dockerfile.py:309: libmodulemd not available
7886
SKIPPED [1] tests/plugins/test_koji_import.py:1570: libmodulemd not available
7887
SKIPPED [1] tests/plugins/test_koji_parent.py:118: Raising for manifests_mismatches is disabled
7888
=========== 9699 passed, 21 skipped, 5746 warnings in 494.81 seconds ===========
```

**Comment by SalatskySal** [comment]
> Can we address skipped tests in centos8 job?
> 
> ```
> =========================== short test summary info ============================
> 7881
> SKIPPED [5] tests/plugins/test_flatpak_create_dockerfile.py:91: libmodulemd not available
> 7882
> SKIPPED [7] tests/plugins/test_flatpak_create_oci.py:467: libmodulemd not available
> 7883
> SKIPPED [1] tests/plugins/test_flatpak_create_oci.py:700: libmodulemd not available
> 7884
> SKIPPED [5] tests/plugins/test_flatpak_update_dockerfile.py:172: libmodulemd not available
> 7885
> SKIPPED [1] tests/plugins/test_flatpak_update_dockerfile.py:309: libmodulemd not available
> 7886
> SKIPPED [1] tests/plugins/test_koji_import.py:1570: libmodulemd not available
> 7887
> SKIPPED [1] tests/plugins/test_koji_parent.py:118: Raising for manifests_mismatches is disabled
> 7888
> =========== 9699 passed, 21 skipped, 5746 warnings in 494.81 seconds ===========
> ```

I suppose we cannot. 'python-libmodulemd' is not available for CentOS 8

**Comment by MartinBasti** [comment]
Issues
```
MANIFEST.in:include requirements-py2.txt   # file was removed
atomic_reactor/plugins/pre_add_filesystem.py:9:    # py2  # whole py2 block not needed

# can be replaced with something py3 sane
atomic_reactor/plugins/pre_add_filesystem.py:161:        config.readfp(config_fp)     # pylint: disable=deprecated-method; py2 compatibility

# can be removed and the whole py2 import block at the top
atomic_reactor/utils/yum.py:98:                    # Fall back to python2 method
```

Dir `images/dockerhost-builder/*` uses py2 packages on latest fedora. @rcerven what's purpose of it? Can we drop it?

**Comment by rcerven** [comment]
> Issues
> 
> ```
> MANIFEST.in:include requirements-py2.txt   # file was removed
> atomic_reactor/plugins/pre_add_filesystem.py:9:    # py2  # whole py2 block not needed
> 
> # can be replaced with something py3 sane
> atomic_reactor/plugins/pre_add_filesystem.py:161:        config.readfp(config_fp)     # pylint: disable=deprecated-method; py2 compatibility
> 
> # can be removed and the whole py2 import block at the top
> atomic_reactor/utils/yum.py:98:                    # Fall back to python2 method
> ```
> 
> Dir `images/dockerhost-builder/*` uses py2 packages on latest fedora. @rcerven what's purpose of it? Can we drop it?

those are used in test_cli & test_buildimage

**Review by MartinBasti — APPROVED** [review]

**Comment by MartinBasti** [comment]
Let's address that test in another PR if needed

File: tests/plugins/test_push_floating_tags.py [hunk 3]
```diff
@@ -34,17 +33,17 @@
...
+        return bytes(value, 'utf-8')
 
 
 def to_text(value):
-    if isinstance(value, text_type):
+    if isinstance(value, str):
         return value
     else:
-        return text_type(value, 'utf-8')
+        return str(value)
```
Comment (tkdchen) [inline tests/plugins/test_push_floating_tags.py:46]: Missing argument `encoding`? Or just call `value.encode()`?

Comment (SalatskySal) [reply tests/plugins/test_push_floating_tags.py:46]: Good catch, thanks!

**Comment by tkdchen** [comment]
LGTM

**Review by tkdchen — APPROVED** [review]

**Comment by MartinBasti** [comment]
Release notes
```
# Incompatible change
- Python2 support removed
```

**Pull request merged**